examples/prom/maxwell_local_rom_greedy.cpp

```diff
@@ -0,0 +1,725 @@
+                   "--no-partial-assembly", "Enable Partial Assembly.");
+    args.AddOption(&device_config, "-d", "--device",
+                   "Device configuration string, see Device::Configure().");
+    args.AddOption(&visit, "-visit", "--visit-datafiles", "-no-visit",
+                   "--no-visit-datafiles",
+                   "Save data files for VisIt (visit.llnl.gov) visualization.");
+    args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                   "--no-visualization",
+                   "Enable or disable GLVis visualization.");
+#ifdef MFEM_USE_AMGX
+    args.AddOption(&useAmgX, "-amgx", "--useAmgX", "-no-amgx",
+                   "--no-useAmgX",
+                   "Enable or disable AmgX in MatrixFreeAMS.");
+#endif
+    args.AddOption(&fom, "-fom", "--fom", "-no-fom", "--no-fom",
+                   "Enable or disable the fom phase.");
+    args.AddOption(&offline, "-offline", "--offline", "-no-offline","--no-offline",
+                   "Enable or disable the offline phase.");
+    args.AddOption(&online, "-online", "--online", "-no-online", "--no-online",
+                   "Enable or disable the online phase.");
+    args.AddOption(&build_database, "-build_database", "--build_database",
+                   "-no-build_database", "--no-build_database",
+                   "Enable or disable the build_database phase of the greedy algorithm.");
+    args.AddOption(&use_database, "-use_database", "--use_database",
+                   "-no-use_database", "--no-use_database",
+                   "Enable or disable the use_database phase of the greedy algorithm.");
+    args.AddOption(&greedy_param_space_min, "-greedy-param-min",
+                   "--greedy-param-min", "The minimum value of the parameter point space.");
+    args.AddOption(&greedy_param_space_max, "-greedy-param-max",
+                   "--greedy-param-max", "The maximum value of the parameter point space.");
+    args.AddOption(&greedy_param_space_size, "-greedy-param-size",
+                   "--greedy-param-size",
+                   "The number of values to search in the parameter point space.");
+    args.AddOption(&greedy_relative_error_tol, "-greedyrelerrortol",
+                   "--greedyrelerrortol", "The greedy algorithm relative error tolerance.");
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+
+    args.Parse();
+    if (!args.Good())
+    {
+        if (myid == 0)
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (myid == 0)
+    {
+        args.PrintOptions(cout);
+    }
+
+    if (fom)
+    {
+        MFEM_VERIFY(fom && !offline
+                    && !online, "everything must be turned off if fom is used.");
```

Comment:
This message could be a little clearer, by writing something like "The FOM phase cannot be run with the offline or online ROM phase."